### PR TITLE
Added in memory collector

### PR DESCRIPTION
**Pull request by baartosz**
Hi. 
I needed to measure processes' metrics and retrieve them in same request/run and hook it up with existing storing procedures. It will be useful to stick to same interface later when we will move that outside to graphite or something. I think it may be useful for others too in some cases.

### CODE REVIEW

File: src/Beberlei/Metrics/Collector/InMemory.php [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    {
+        $this->gauge($variable, $time);
+    }
+
+    /**
+     * Sends the metrics to the adapter backend.
+     */
+    public function flush()
+    {
+        // TODO: Implement flush() method.
```
Comment (lyrixx) [inline src/Beberlei/Metrics/Collector/InMemory.php]: You forgot this TODO. IMHO, you could juste reset the current state.

File: src/Beberlei/Metrics/Tests/Collector/InMemoryTest.php [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+    /** @var  InMemory */
+    private $collector;
+
+    public function setUp()
+    {
+        $this->collector = new InMemory();
+    }
+
+    public function test_increment()
```
Comment (lyrixx) [inline src/Beberlei/Metrics/Tests/Collector/InMemoryTest.php]: We are using camelCase coding style.

**Comment by baartosz** [comment]
Thanks for feedback. Updated PR. 
A bit tricky is that gauge cannot be set to negative value without setting it to 0 first according to 
https://github.com/etsy/statsd/blob/master/docs/metric_types.md#gauges

**Comment by lyrixx** [comment]
Thanks for your work on this new feature!

**Pull request closed without merging**